src/fixed_rational.jl

```diff
@@ -0,0 +1,30 @@
+"""
+    FixedRational{T,den}
+
+A rational number with a fixed denominator. Significantly
+faster than `Rational{T}`, as it never needs to compute
+the `gcd` apart from when printing.
+"""
+struct FixedRational{T<:Integer,den} <: Real
+    num::T
+    global unsafe_fixed_rational(num::Integer, ::Type{T}, ::Val{den}) where {T,den} = new{T,den}(num)
```

Comment:
why not just define this outside the `struct`?

Reply:
It uses `new` so needs to be inside.

The reason for this separate function is that defining `FixedRational{T,den}(::Integer)` was ambiguous with the original type definition

Reply:
right. Makes sense.